android/src/main/java/io/customer/reactnative/sdk/CustomerIOReactNativeModule.kt

```diff
@@ -1,105 +1,177 @@
-    private val application: Application = reactApplicationContext.applicationContext as Application
+    reactContext: ReactApplicationContext,
+) : ReactContextBaseJavaModule(reactContext) {
+    private val logger: Logger = CustomerIOReactNativeInstance.logger
     private lateinit var customerIO: CustomerIO
 
-    override fun getName(): String {
-        return "CustomerioReactnative"
+    init {
+        val isDebuggable = 0 != reactContext.applicationInfo.flags and
+                ApplicationInfo.FLAG_DEBUGGABLE
+        CustomerIOReactNativeInstance.setLogLevel(
+            logLevel = if (isDebuggable) CioLogLevel.DEBUG else CioLogLevel.ERROR,
+        )
     }
 
-    fun logMessage(message: String) {
-        Log.d("[CIO]", message)
+    override fun getName(): String {
+        return MODULE_NAME
     }
 
-    private fun isInitialized(): Boolean {
+    private fun isInstanceValid(): Boolean {
         return ::customerIO.isInitialized
     }
 
-    private fun isConfigurationsPending(): Boolean {
-        val isPending = !isInitialized()
-        if (isPending) {
-            logMessage("Customer.io instance not initialized")
+    private fun isNotInitialized(): Boolean {
+        val isInstanceInvalid = !isInstanceValid()
+        if (isInstanceInvalid) {
+            logger.error("Customer.io instance not initialized")
         }
-        return isPending
+        return isInstanceInvalid
     }
 
     @JvmOverloads
     @ReactMethod
-    fun initialize(siteId: String, apiKey: String, region: String? = null) {
-        if (isInitialized()) return
-
-        customerIO = CustomerIO.Builder(
-            siteId = siteId,
-            apiKey = apiKey,
-            region = region.toRegion() ?: Region.US,
-            appContext = application,
-        )
-            .setLogLevel(CioLogLevel.DEBUG)
-            .autoTrackDeviceAttributes(true)
-            .build()
+    fun initialize(
+        environment: ReadableMap,
+        configuration: ReadableMap? = null,
+        sdkVersion: String? = null,
+    ) {
+        if (isInstanceValid()) {
+            logger.info("Customer.io instance already initialized")
+            return
+        }
 
+        val env = environment.toMap()
+        val config = configuration?.toMap()
+
+        try {
+            val siteId = env.getString(Keys.Environment.SITE_ID)
+            val apiKey = env.getString(Keys.Environment.API_KEY)
+            val region = env.getProperty<String>(
+                Keys.Environment.REGION
+            )?.takeIfNotBlank().toRegion()
+            val organizationId = env.getProperty<String>(
+                Keys.Environment.ORGANIZATION_ID
+            )?.takeIfNotBlank()
+
+            customerIO = CustomerIO.Builder(
+                siteId = siteId,
+                apiKey = apiKey,
+                region = region,
+                appContext = reactApplicationContext.applicationContext as Application,
```

Comment:
Just curious to understand what `appContext` does ?

Reply:
`appContext` is `Context` instance which is the core class of Android. It lets you assess resources, identify build types, start screens, and anything that is directly linked with Android OS 🤓